### PR TITLE
upgrade min deps

### LINKED ISSUE #4928: update min dependencies
## 🐛 Bug

after pip install strategy changes we have several packages which are not feasible to install together

### To Reproduce

see all min CI config https://github.com/PyTorchLightning/pytorch-lightning/runs/1478280229

### Expected behavior

update some libs to be installable, probably go one by one or list dependencies of our dependencies and find a minimal intersection

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

Fixes #4928 

`onnxruntime 1.3` requires `numpy>=1.16.6`
`gym 0.17` requires `cloudpickle>=1.3`
`hydra-core 1.0` requires `omegaconf>=2.0.1`

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by ydcjeff** [comment]
I think macOS fail is unrelated? It passed on linux and windows tho.

**Review by Borda — APPROVED** [review]
Great!

File: requirements/test.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 coverage>=5.0
 codecov>=2.1
 pytest>=5.0
-pytest-cov
-pytest-flake8
+# pytest-cov
```
Comment (SeanNaren) [inline requirements/test.txt:4]: Any particular reason why these are commented out?

Comment (Borda) [reply requirements/test.txt:4]: I did it now bc Mac was complaining and we do need it anymore

**Comment by Borda** [comment]
seems there is some more for macOS
```
ERROR: Cannot install -r ./requirements/./examples.txt (line 1), -r ./requirements/./extra.txt (line 8) and torch==1.4 because these package versions have conflicting dependencies.
```

**Comment by ydcjeff** [comment]
> seems there is some more for macOS
> 
> ```
> ERROR: Cannot install -r ./requirements/./examples.txt (line 1), -r ./requirements/./extra.txt (line 8) and torch==1.4 because these package versions have conflicting dependencies.
> ```

torch 1.4 requires torchvision and torchtext 0.5

**Comment by Borda** [comment]
this looks like some connection error
```
ERROR: HTTP error 403 while getting https://download.pytorch.org/whl/torchtext-0.5.0-py3-none-any.whl (from https://download.pytorch.org/whl/cpu/torch_stable.html)
```
as the file seems to exists
Screenshot 2020-12-01 at 16 13 17

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**